packages/components/PrefixBadgePicker/PrefixBadgePicker.tsx

```diff
@@ -29,153 +21,57 @@ export function PrefixBadgePicker({
+            variants={{
+              active: { scale: 1 },
+              opened: { scale: 1 },
+              closed: { scale: 0.82 },
             }}
+            className={styles.PrefixBadge}
           >
-            <motion.span
-              animate={{
-                width: isOpen ? "0.5rem" : "0.3rem",
-                height: isOpen ? "0.1rem" : "0.3rem",
-              }}
-              transition={{ type: "tween", duration: 0.16 }}
-              className={styles.DefaultBadge}
+            <motion.div
+              className={
+                badge === "unicorn"
+                  ? styles.UnicornBadge
+                  : classNames(styles.ColorBadge, styles[badge])
+              }
             />
           </motion.button>
-        )}
-
-        {badges.map(badge => {
-          const sharedProps = {
-            key: badge,
-            layoutId: id + badge,
-            name: badge,
-            onKeyDown: (event: React.KeyboardEvent) => {
-              if (event.key === "Enter") {
-                if (state === "closed") {
-                  setState("opening");
-                } else {
-                  handleSelect(badge);
-                }
-              }
-            },
-            onMouseDown: () => {
-              if (state === "closed") {
-                setState("opening");
-              }
-            },
-            onMouseUp: () => {
-              if (isOpen) {
-                handleSelect(badge);
-              } else {
-                setState("opened");
-              }
-            },
-          };
-          const isActive = badge === activeBadge;
-
-          return isOpen ? (
-            <PrefixBadgeItem {...sharedProps} isActive={isActive} />
-          ) : isActive ? (
-            <PrefixBadgeItem {...sharedProps} />
-          ) : null;
-        })}
-      </AnimatePresence>
-    </motion.div>
+        ))}
+      </Picker>
+    </div>
   );
 }
 
-function PrefixBadgeItem({
-  isActive,
-  name,
-  layoutId,
-  ...props
-}: {
-  isActive?: boolean;
-  name: PrefixBadge;
-  layoutId: string;
-} & Omit<React.ComponentProps<typeof motion.button>, "onSelect">) {
+function ToggleButton(props: React.ComponentProps<typeof motion.button>) {
   return (
-    <motion.button
-      type="button"
-      layoutId={layoutId}
-      className={classNames(styles.PrefixBadge, { [styles.active]: isActive })}
-      initial={{ opacity: 0, scale: 0.9 }}
-      animate={{ opacity: 1, scale: 1 }}
-      exit={{ opacity: 0, scale: 0.9 }}
-      transition={{ duration }}
-      {...props}
-    >
-      <div
-        className={
-          name === "unicorn" ? styles.UnicornBadge : classNames(styles.ColorBadge, styles[name])
-        }
+    <motion.button {...props} className={classNames(styles.PrefixBadge, props.className)}>
+      <motion.span
+        layout
+        variants={{
+          opened: { width: "0.5rem", height: "0.1rem" },
+          closed: { width: "0.3rem", height: "0.3rem" },
+        }}
```

Comment:
Variants are a pattern I want to explore in our design system since it works well with composition like they're being used here. It's a simple idea of a parent (Picker in this case) being able to set which variants are active and then cascading down so children can describe their state at that time. This is similar to the classnames package, but would be a first-class feature of our design system and would include JS animations when needed.